web/react/components/create_comment.jsx

```diff
@@ -106,13 +106,23 @@ module.exports = React.createClass({
         }
     },
     handleUserInput: function(messageText) {
+        var newPostError = this.state.postError;
+
+        var len = utils.getLengthOfTextInTextarea(messageText);
+
+        if (!this.state.postError && len >= Constants.MAX_POST_LEN) {
+            newPostError = 'Comment length cannot exceed 4000 characters';
```

Comment:
The error message is slightly different depending on the situation ("Message length...", "Comment length...", and "New message length...") but your right I should at least set it as a variable within each component (and like @nickago said I should be using MAX_POST_LEN, good catch, made the dang thing for that exact reason :P ).  I was initially hesitant to refactor the whole thing into its own separate function with the differences in the error message and the size of the code, but I'll go ahead and see about moving it into its own function.